### DISCUSSION TITLE
i18n on extensions

### DISCUSSION THREAD

**Discussion by naskio** [directus/directus#12185] [Archive]
Hello everyone
I am wondering if there is a way to use translated strings when building extensions (also where I can add new strings and the corresponding translations).
For example:
```
name: '$t:extensions.displays.previewdocs.name',
```
Thanks

**Comment by azrikahar** [comment]
What you did should work, but of course currently it'll only work for existing translation keys in [en-US.yaml](https://github.com/directus/directus/blob/main/app/src/lang/translations/en-US.yaml).

With that said, the good news is you will be able to add new strings and the corresponding translations once #12170 lands 👍

**Reply by massimoliani** [reply]
Few questions:
1. Is there any documentation or example on using i18n (connected to new translation string collection) inside custom interfaces? 
2. Is there a way to create translation items from the app (instedad of manually writing key value the interface create first set of key inside translation collection and user later will translate it).
Thank you.

**Comment by programmarchy** [comment]
Is there any way to define new translation keys for an extension that gets loaded with the extension?

For example, I built a custom operation, and wanted to use translation keys to define strings like the operation description, field names, field notes, etc. like built-in operations do.

Ideally, Directus would pick up translations(e.g. defined in `lang/<lang>.yaml` in the extension's source folder) that get loaded along with the extension.
